dashboard/modules/job/data_types.py

```diff
@@ -1,11 +1,6 @@
 from enum import Enum
 from typing import Any, Dict
-
-try:
-    from pydantic import BaseModel
-except ImportError:
-    # Lazy import without breaking class def
-    BaseModel = object
```

Comment:
exiting tests actually won't if user don't have pydantic installed since python object and BaseModel have different APIs to create and serialize objects. Because we can't add pydantic to oss yet due to its size (30MB), this PR falls back to dataclass.